### PR TITLE
Core functionality project actions [QA-76]

### PR DESCRIPTION
**Pull request by laurenrevere**
## Purpose
Add the `core_functionality` tests for an OSF project.

On the side did some cleanup/maintenance.

Threw in another change: Started using solely environment variables.

## Summary of Changes
- Add basic project tests (confirm files load, you can change the title, and make a project public/private)
- Handle logged out views for unauthenticated users. When you visit a private page as a logged out human, the OSF redirects you to CAS. Added the `expect_login_redirect` to `goto` so that it can handle those cases gracefully. (Otherwise it would fail when it attempts to verify) 
- Hide the cookie banner (so it doesn't get in the way of clicking other elements)
- Add the `qatest` tag to projects by default (used to only add it if there were plans to make the project public)

## Testing Changes Moving Forward
- If someone wants to ensure a project they created will show up in search, will need to explicitly create a project with no `qatest` tag.
- If someone wants to test the cookie banner will need to remove the setting of its acceptance cookie that now automatically happens.
- Make sure to watch out for caching problems. Especially server caching problems.
- IMPORTANT: Because of envars changes, can decide which server to run travis on by changing environment variables there instead of making a PR. Yay. Long time overdue and h/t @sloria. 

## Ticket

https://openscience.atlassian.net/browse/QA-76

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        'api': 'https://osf-api.io',
+        'files': 'https://files.us.osf.io',
+        'cas': 'https://accounts.osf.io'
+    }
+}
+
+DRIVER = os.environ.get('DRIVER', 'Firefox')
+HEADLESS = os.environ.get('HEADLESS', False)
+
+QUICK_TIMEOUT = os.environ.get('QUICK_TIMEOUT', 4)
```
Comment (sloria) [inline settings.py]: Drive-by: environment variables will always be strings, `os.environ` won't do any sort of type-casting for you. You would have to do something like

```python
QUICK_TIMEOUT = int(os.environ.get('QUICK_TIMEOUT', 4))
```

You could use a [environs](https://github.com/sloria/environs), which will do type-casting for you:

```python
from environs import Env
env = Env()
env.read_env()

QUICK_TIMEOUT = env.int('QUICK_TIMEOUT', default=4)
```

It'll also allow you to have required envvars. Just omit the `default` argument.

```python
# no default given, so this will error if USER_ONE_PASSWORD
# is unset (that's a good thing)
USER_ONE_PASSWORD = env.str('USER_ONE_PASSWORD')
```

Full disclosure: I'm the author of environs.

Comment (laurenrevere) [reply settings.py]: Ahhh good catch. Thanks, Steve!
